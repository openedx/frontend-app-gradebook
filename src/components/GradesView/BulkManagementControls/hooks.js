import { actions, selectors } from 'data/redux/hooks';

export const useBulkManagementControlsData = () => {
  const gradeExportUrl = selectors.root.useGradeExportUrl();
  const showBulkManagement = selectors.root.useShowBulkManagement();
  const downloadBulkGradesReport = actions.grades.useDownloadBulkGradesReport();

  const handleClickExportGrades = () => {
    downloadBulkGradesReport();
    window.location.assign(gradeExportUrl);
  };

  return {
    show: showBulkManagement,
    handleClickExportGrades,
  };
};
export default useBulkManagementControlsData;
