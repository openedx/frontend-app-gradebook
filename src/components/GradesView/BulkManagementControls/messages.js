import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  downloadGradesBtn: {
    id: 'gradebook.GradesView.BulkManagementControls.bulkManagementLabel',
    defaultMessage: 'Download Grades',
    description: 'A labeled button that allows an admin user to download course grades all at once (in bulk).',
  },
});

export default messages;
