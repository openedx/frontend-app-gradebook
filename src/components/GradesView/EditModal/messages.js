import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignmentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.assignment',
    defaultMessage: 'Assignment',
    description: 'Edit Modal Assignment header',
  },
  currentGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.currentGrade',
    defaultMessage: 'Current Grade',
    description: 'Edit Modal Current Grade header',
  },
  originalGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.originalGrade',
    defaultMessage: 'Original Grade',
    description: 'Edit Modal Original Grade header',
  },
  studentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.student',
    defaultMessage: 'Student',
    description: 'Edit Modal Student header',
  },
  title: {
    id: 'gradebook.GradesView.EditModal.title',
    defaultMessage: 'Edit Grades',
    description: 'Edit Modal title',
  },
  closeText: {
    id: 'gradebook.GradesView.EditModal.closeText',
    defaultMessage: 'Cancel',
    description: 'Edit Modal close button text',
  },
  visibility: {
    id: 'gradebook.GradesView.EditModal.contactSupport',
    defaultMessage: 'Showing most recent actions (max 5). To see more, please contact support',
    description: 'Edit Modal visibility hint message',
  },
  saveVisibility: {
    id: 'gradebook.GradesView.EditModal.saveVisibility',
    defaultMessage: 'Note: Once you save, your changes will be visible to students.',
    description: 'Edit Modal saved changes effect hint',
  },
  saveGrade: {
    id: 'gradebook.GradesView.EditModal.saveGrade',
    defaultMessage: 'Save Grades',
    description: 'Edit Modal Save button label',
  },
});

export default messages;
