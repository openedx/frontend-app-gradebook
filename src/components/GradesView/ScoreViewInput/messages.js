import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  scoreView: {
    id: 'gradebook.GradesView.scoreViewLabel',
    defaultMessage: 'Score View',
    description: 'The label for the dropdown list that allows a user to select the Score format',
  },
  absolute: {
    id: 'gradebook.GradesView.absoluteOption',
    defaultMessage: 'Absolute',
    description: 'A label within the Score Format dropdown list for the Absolute Grade Score option',
  },
  percent: {
    id: 'gradebook.GradesView.percentOption',
    defaultMessage: 'Percent',
    description: 'A label within the Score Format dropdown list for the Percent Grade Score option',
  },
});

export default messages;
