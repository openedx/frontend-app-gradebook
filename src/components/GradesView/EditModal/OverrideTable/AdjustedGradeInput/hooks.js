import { actions, selectors } from 'data/redux/hooks';
import { getLocalizedSlash } from 'i18n/utils';

const useAdjustedGradeInputData = () => {
  const possibleGrade = selectors.root.useEditModalPossibleGrade();
  const value = selectors.app.useModalData().adjustedGradeValue;
  const setModalState = actions.app.useSetModalState();
  const hintText = possibleGrade && ` ${getLocalizedSlash()} ${possibleGrade}`;

  const onChange = ({ target }) => {
    setModalState({ adjustedGradeValue: target.value });
  };

  return {
    value,
    onChange,
    hintText,
  };
};

export default useAdjustedGradeInputData;
