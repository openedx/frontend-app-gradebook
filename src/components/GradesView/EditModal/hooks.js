import { selectors, actions, thunkActions } from 'data/redux/hooks';

export const useEditModalData = () => {
  const error = selectors.grades.useGradeData().gradeOverrideHistoryError;
  const isOpen = selectors.app.useModalData().open;
  const closeModal = actions.app.useCloseModal();
  const doneViewingAssignment = actions.grades.useDoneViewingAssignment();
  const updateGrades = thunkActions.grades.useUpdateGrades();

  const onClose = () => {
    doneViewingAssignment();
    closeModal();
  };

  const handleAdjustedGradeClick = () => {
    updateGrades();
    doneViewingAssignment();
    closeModal();
  };

  return {
    onClose,
    error,
    handleAdjustedGradeClick,
    isOpen,
  };
};

export default useEditModalData;
