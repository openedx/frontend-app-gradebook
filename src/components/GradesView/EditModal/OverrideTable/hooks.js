import { useIntl } from '@edx/frontend-platform/i18n';

import { gradeOverrideHistoryColumns as columns } from 'data/constants/app';
import { selectors } from 'data/redux/hooks';

import messages from './messages';

const useOverrideTableData = () => {
  const { formatMessage } = useIntl();

  const hide = selectors.grades.useHasOverrideErrors();
  const gradeOverrides = selectors.grades.useGradeData().gradeOverrideHistoryResults;
  const tableProps = {};
  if (!hide) {
    tableProps.columns = [
      { Header: formatMessage(messages.dateHeader), accessor: columns.date },
      { Header: formatMessage(messages.graderHeader), accessor: columns.grader },
      { Header: formatMessage(messages.reasonHeader), accessor: columns.reason },
      { Header: formatMessage(messages.adjustedGradeHeader), accessor: columns.adjustedGrade },
    ];
    tableProps.data = gradeOverrides;
  }
  return { hide, ...tableProps };
};

export default useOverrideTableData;
