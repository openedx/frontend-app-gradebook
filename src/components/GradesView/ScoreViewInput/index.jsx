import React from 'react';

import { Form } from '@edx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import { actions, selectors } from 'data/redux/hooks';
import messages from './messages';

/**
 * <ScoreViewInput />
 * redux-connected select control for grade format (percent vs absolute)
 */
export const ScoreViewInput = () => {
  const { formatMessage } = useIntl();
  const { gradeFormat } = selectors.grades.useGradeData();
  const toggleFormat = actions.grades.useToggleGradeFormat();
  return (
    <Form.Group controlId="ScoreView">
      <Form.Label>{formatMessage(messages.scoreView)}:</Form.Label>
      <Form.Control
        as="select"
        value={gradeFormat}
        onChange={toggleFormat}
      >
        <option value="percent">{formatMessage(messages.percent)}</option>
        <option value="absolute">{formatMessage(messages.absolute)}</option>
      </Form.Control>
    </Form.Group>
  );
};
ScoreViewInput.propTypes = {};

export default ScoreViewInput;
