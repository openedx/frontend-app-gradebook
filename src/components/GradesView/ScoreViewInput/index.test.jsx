import React from 'react';
import { shallow } from 'enzyme';

import { useIntl } from '@edx/frontend-platform/i18n';
import { GradeFormats } from 'data/constants/grades';

import { formatMessage } from 'testUtils';
import { actions, selectors } from 'data/redux/hooks';
import ScoreViewInput from '.';
import messages from './messages';

jest.mock('data/redux/hooks', () => ({
  actions: {
    grades: { useToggleGradeFormat: jest.fn() },
  },
  selectors: {
    grades: { useGradeData: jest.fn() },
  },
}));

const toggleGradeFormat = jest.fn().mockName('hooks.toggleGradeFormat');
actions.grades.useToggleGradeFormat.mockReturnValue(toggleGradeFormat);
const gradeFormat = 'test-grade-format';
selectors.grades.useGradeData.mockReturnValue({ gradeFormat });

let el;
describe('ScoreViewInput component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    el = shallow(<ScoreViewInput />);
  });
  describe('behavior', () => {
    it('initializes intl hook', () => {
      expect(useIntl).toHaveBeenCalled();
    });
    it('initializes redux hooks', () => {
      expect(actions.grades.useToggleGradeFormat).toHaveBeenCalled();
      expect(selectors.grades.useGradeData).toHaveBeenCalled();
    });
  });
  describe('render', () => {
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });
    test('label', () => {
      const label = el.children().at(0);
      expect(label.text()).toEqual(`${formatMessage(messages.scoreView)}:`);
    });
    describe('form control', () => {
      let control;
      beforeEach(() => {
        control = el.children().at(1);
      });
      test('value and onChange from redux hooks', () => {
        expect(control.props().value).toEqual(gradeFormat);
        expect(control.props().onChange).toEqual(toggleGradeFormat);
      });
      test('absolute and percent options', () => {
        const children = control.children();
        expect(children.at(0).props().value).toEqual(GradeFormats.percent);
        expect(children.at(0).text()).toEqual(formatMessage(messages.percent));
        expect(children.at(1).props().value).toEqual(GradeFormats.absolute);
        expect(children.at(1).text()).toEqual(formatMessage(messages.absolute));
      });
    });
  });
});
