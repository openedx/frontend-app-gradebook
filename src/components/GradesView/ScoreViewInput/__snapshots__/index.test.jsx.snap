// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreViewInput component render snapshot 1`] = `
<Form.Group
  controlId="ScoreView"
>
  <Form.Label>
    Score View
    :
  </Form.Label>
  <Form.Control
    as="select"
    onChange={[MockFunction hooks.toggleGradeFormat]}
    value="test-grade-format"
  >
    <option
      value="percent"
    >
      Percent
    </option>
    <option
      value="absolute"
    >
      Absolute
    </option>
  </Form.Control>
</Form.Group>
`;
